### ISSUE TITLE
[FR] Track watch status and progress of downloaded videos

### ISSUE THREAD

**Issue by woj-tek** [TeamNewPipe/NewPipe#3094]
- [X] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [X] I checked if the issue/feature exists in the latest version.

Currently, because NewPipe uses external application to play downloaded videos, it doesn't track playback progress nor watch status of those. It would be nice to actually  at least mark that we watched certain video.

I'm not sure if this is not in a way duplicate of #478 - probably having option to play those with internal player would solve that issue.

**Comment by john3voltas** [comment]
This is really a must have.
It's a big mess when I'm going through my channels/playlists trying to identify which episodes I've watched and which new and still unwatched.
There should be a way for us to mark those seen, even if it had to be marked by hand. Better something than nothing...

**Comment by gkeegan** [comment]
@john3voltas you can go into the video and just set the watch time to about 90% through the video to get the red "progress bar" on the thumbnail. But i agree that a better method would be nice.

**Comment by john3voltas** [comment]
Hi @gkeegan 
That's a workaround but it doesn't make much sense. After watching, the red progress bar should stay and not reset itself as it seems that it is currently doing...

**Comment by gkeegan** [comment]
Yeah I know it's a workaround but it works for now until something else gets implemented. And I agree that would be proper behavior.

**Comment by opusforlife2** [comment]
I don't see why Newpipe should do this. VLC has this feature.

If someone _does_ want to implement this: dependent on #478.

**Comment by SameenAhnaf** [comment]
@litetex Out of scope as #478 is rejected
